### PR TITLE
adding a some hook

### PR DESCRIPTION
**Pull request by ekryski**
### Summary

Adding a utility hook that takes in other functions and resolves to `true` if any of the functions return true. Resolves to `false` if all functions return falsey values. Errors thrown from a hook function are deemed as falsey and are swallowed. This is the only thing I'm not sure of. What should it do?

**EDIT**: It now rejects errors in the hook chain properly.

Usage is like so:

```js
some(hook1, hook2, hook3)
````

### Other Information

Related to this discussion https://github.com/feathersjs/feathers-permissions/issues/4

cc/ @daffl @bedeoverend

### CODE REVIEW

**Comment by bedeoverend** [comment]
@ekryski I'd vote no on swallowing errors, at least by default. In the context of [feathersjs/feathers-permissions#4](https://github.com/feathersjs/feathers-permissions/issues/4) it might make sense to swallow, given that I'd imagine permissions would deny on failure, but as a generic hook, I'd prefer it to crash so I could handle _unexpected_ errors.

Perhaps it should swallow only errors with status codes `40x`? i.e. implying that the error is expected, but that might be getting a little too magical.

File: src/new.js [hunk 1]
```diff
@@ -178,6 +178,44 @@ export const iff = (ifFcn, ...rest) => {
...
+ *
+ * Example 2 - called within a custom hook function
+ * function (hook) {
+ *   ...
+ *   return hooks.some(hook1, hook2, ...).call(this, currentHook)
+ *     .then(hook => { ... });
+ * }
+ */
+
+export const some = (...rest) => (hook) => {
```
Comment (bedeoverend) [inline src/new.js]: I think the inner arrow function `(hook) => {` should be `function(hook) {` so the right context is given for `fn.call(this, ...)` later on

**Comment by ekryski** [comment]
Ya I was thinking that if an error happens you'd probably want to know about it as it's unexpected behaviour and it should error out of the `some` hook chain rather than swallow it. Unless anyone has any issues with that I'm going to update it so that it doesn't treat errors as falsey and swallow them.

**Comment by ekryski** [comment]
I dunno what code climate is complaining about but I can't access the repo in there to get more details. Just get a 422 error. Everything else is all good.

**Comment by ekryski** [comment]
ok I'm shipping it. I need this to wrap permissions.

**Comment by eddyystop** [comment]
Sorry for not getting onto this. I'm preparing for a business supper.

**Comment by ekryski** [comment]
np. I think it's pretty safe.

**Comment by eddyystop** [comment]
GitHub shows on the PR that Code Climate failed. You can click a link on that line to see the CC page and that eventually leads you to the issues it found.

**Comment by eddyystop** [comment]
Needs tests.

**Comment by ekryski** [comment]
> GitHub shows on the PR when Code Climate failed. You can click a link on that line to see the CC page and that eventually leads you to the issues it found.

I just get this. Must be a permissions thing.

screen shot 2016-12-08 at 7 03 15 pm

> Needs tests.

Has tests and they are passing 😄

**Pull request merged**